micromanager_gui/main_window.py

```diff
@@ -76,6 +76,8 @@ def __init__(self, viewer: napari.viewer.Viewer, remote=False):
         sig.stagePositionChanged.connect(self._on_stage_position_changed)
         sig.exposureChanged.connect(self._on_exp_change)
 
+        sig.imageSnapped.connect(self._snap)
```

Comment:
No need to update the viewer in a thread, just pass directly to the update function

```suggestion
        sig.imageSnapped.connect(self.update_viewer)
```